modules/task/backend.go

```diff
@@ -101,22 +93,20 @@ func (b *inMemBackend) Name() string {
 }
 
 // Start implements the Module interface
-func (b *inMemBackend) Start(ready chan<- struct{}) <-chan error {
-	errorCh := make(chan error, 2)
-	if b.IsRunning() {
-		errorCh <- errors.New("cannot start when module is already running")
-	} else if b.state == _stopped {
-		errorCh <- errors.New("cannot start when module has been stopped")
-	} else {
-		b.state = _running
-		go b.consumeFromQueue(errorCh)
-	}
-	return errorCh
+func (b *inMemBackend) Start() error {
+	go b.consumeFromQueue()
+	return nil
 }
 
-func (b *inMemBackend) consumeFromQueue(errorCh chan error) {
+func (b *inMemBackend) consumeFromQueue() {
 	for msg := range b.bufQueue {
-		errorCh <- Run(context.Background(), msg)
+		// TODO(pedge): this was effectively not being handled and was a bug
```

Comment:
It is not a bug, but a feature :) Seriously, this is a way to communicate with a sender about errors in function executions.